File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        st = ""
        for i in strs:
            st += str(len(i))+ "#" + i
        return st

    def decode(self, s: str) -> List[str]:

        result = []
        i = 0
        while i < len(s):
            j = i
            while s[j]!= "#":
                j += 1
            length = int(s[i:j])
            i = j+ 1
            j = i +length
            result.append(s[i:j])
            i = j
        return result
            

File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()){
         return "";
        }
        string result = "";
       sort(strs.begin(),strs.end());
       /* comparing the first and the last element to get the biggest continous array */
       
       string s_first = strs[0];
       string s_last = strs.back();

      
       for(int i=0;i<s_first.size();i++){
        if(s_first[i]!=s_last[i]){
            break;
        }
        else
        result += s_first[i];
       }
       return result;

        }
};